File: Fisher-Yates/PHP/mitogh/fisher_yates.php
<?php
function fisher_yates($a = []){
    $n = count($a);
    for($i = 0; $i < $n; $i++){
        $j = mt_rand(0,$i);
        $tmp = $a[$j];
        $a[$j] = $a[$i];
        $a[$i] = $tmp;
    }
    return $a;
}

File: Fisher-Yates/Objective-C/intoxicated/Fisher-yates.m
//
//  Fisher-yates.m
//  Algorithms
//
//  Created by intoxicated on 1/4/14.
//  Copyright (c) 2014 intoxicated. All rights reserved.
//

#import "Fisher-yates.h"

@implementation Fisher_yates

+ (void)shuffle:(NSMutableArray *)array
{
    int swapIndex;
    id temp;
    
    for(int i = (int)array.count - 1; i > 0; --i)
    {
        swapIndex = arc4random() % i;
        
        temp = [array objectAtIndex:i];
        [array replaceObjectAtIndex:i withObject:[array objectAtIndex:swapIndex]];
        [array replaceObjectAtIndex:swapIndex withObject:temp];
    }
}

@end
